### PR TITLE
Remove the redundant DatasetCache in WorkerContext

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/worker/BasicWorkerContext.java [hunk 1]
```diff
@@ -79,7 +76,7 @@ public BasicWorkerContext(WorkerSpecification spec, Program program, RunId runId
                             DiscoveryServiceClient discoveryServiceClient,
                             StreamWriterFactory streamWriterFactory,
                             @Nullable PluginInstantiator pluginInstantiator) {
-    super(program, runId, runtimeArgs, spec.getDatasets(),
+    super(program, runId, runtimeArgs, ImmutableSet.<String>of(),
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/worker/BasicWorkerContext.java]: Why make this change?
So all datasets accessed will be dynamic then?

**Comment by anew** [comment]
LGTM

**Comment by chtyim** [comment]
Build running in http://builds.cask.co/browse/CDAP-DUT3046

**Comment by chtyim** [comment]
Build passed

**Comment by chtyim** [comment]
Squashed and merging.

**Pull request merged**